src/main/kotlin/no/nav/syfo/personhendelse/PersonhendelseService.kt

```diff
@@ -0,0 +1,56 @@
+package no.nav.syfo.personhendelse
+
+import no.nav.person.pdl.leesah.Endringstype
+import no.nav.person.pdl.leesah.Personhendelse
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.domain.PersonIdentNumber
+import no.nav.syfo.oppfolgingstilfelle.OppfolgingstilfelleService
+import no.nav.syfo.personhendelse.db.addPerson
+import no.nav.syfo.personhendelse.db.getDodsdato
+import no.nav.syfo.util.kafkaCallId
+import org.slf4j.LoggerFactory
+
+class PersonhendelseService(
+    private val database: DatabaseInterface,
+    private val oppfolgingstilfelleService: OppfolgingstilfelleService,
+) {
+    suspend fun handlePersonhendelse(personhendelse: Personhendelse) {
+        if (personhendelse.doedsfall != null && personhendelse.endringstype == Endringstype.OPPRETTET) {
+            log.info("Received personhendelse: Hendelseid: ${personhendelse.hendelseId}, Endringstype: ${personhendelse.endringstype.name}, Oppltype: ${personhendelse.opplysningstype}, Dato: ${personhendelse.doedsfall?.doedsdato}")
+            personhendelse.personidenter
+                .mapNotNull { personIdent ->
+                    try {
+                        if (personIdent.isNullOrEmpty() || personIdent.length != 11) {
+                            null
+                        } else {
+                            PersonIdentNumber(personIdent)
+                        }
+                    } catch (ex: IllegalArgumentException) {
+                        log.warn("Invalid personident for Personhendelse", ex)
+                        null
+                    }
+                }
+                .forEach { personIdent ->
+                    if (isKnownPersonIdent(personIdent)) {
+                        database.connection.use { connection ->
+                            if (connection.getDodsdato(personIdent) == null) {
+                                connection.addPerson(
+                                    uuid = personhendelse.hendelseId,
```

Comment:
Burde vi generere vår egen uuid, evt gjøre UUID.fromString(personhendelse.hendelseId) ?

Reply:
Synes det var fint å bruke uuid'en som kommer med hendelsen her for sporbarhetens del. Men kanskje lurt å gå veien om UUID.fromstring for å validere at det er en korrekt uuid (det bør det jo alltid være når det kommer fra PDL).